### PR TITLE
Issue680 update dependency bestest hydronic

### PR DESCRIPTION
**Pull request by EttoreZ**
This pull request is for #680. Modelica code of `BETEST_hydronic` and `BESTEST_hydronic_heat_pump` was updated to NOT extend from model of IDEAS library, but instead duplicate it, to make test case Modelica code maintenance easier.

### CODE REVIEW

**Comment by EttoreZ** [comment]
One comment. I did not add the new `wrapped.fmu` since the code change does not affect results. This is to avoid further increasing the size repository.

**Comment by dhblum** [comment]
@jelgerjansen This PR proposes to move the code for the ``bestest_hydronic`` and ``bestest_hydronic_heat_pump`` test cases from IDEAS (and included in BOPTEST via Modelica ``extends``), to having the models in the IBPSA repo and have IDEAS as a versioned dependency.  This is the approach taken with other test cases in BOPTEST that depend on IDEAS and Buildings.  We feel this would help decouple maintenance and future development of the models from IDEAS development and maintenance, especially for things that are not IDEAS-specific.  A downside is that the test cases are no longer unit tested with your IDEAS development.  

Let me know if this seems ok to you or if you see reasons we should keep the models within IDEAS, as was set up originally.

Note also that since the model code would be duplicated from and original distribution in IDEAS, I've included reference to the IDEAS license in the BOPTEST license as part of this PR.

**Comment by jelgerjansen** [comment]
Hi @dhblum, I think it is OK to move these models from IDEAS to BOPTEST. According to my knowledge, the main functionalities of both the component models in both models are also tested in other unit tests. Or do I miss something important here @Mathadon?

I noticed that the IDEAS v2.2.1 is referred to, but this was before the update of MSL to v4.0.0. Are you sure about the IDEAS version in the documentation?

**Comment by EttoreZ** [comment]
hey @jelgerjansen answering your second question. This version of the test case still uses Modelica 3.2.3 due to the legacy support of JModelica for compilation, hence IDEAS 2.2.1. We are moving away from JModelica and working to update all BOPTEST test cases to Modelica 4.0 and related version of the libraries (like IDEAS 3.0.0).

**Comment by dhblum** [comment]
Thanks @jelgerjansen.  To your first question about unit tests: BOPTEST unit tests test the compilation and simulation of the full test case models with reference results produced and compared for BOPTEST KPI values and a selection of variable timeseries that are generated during the tests.  I imagine the components of the models are unit tested in the IDEAS library, but you or @Mathadon may be able to confirm better.

**Comment by jelgerjansen** [comment]
@EttoreZ @dhblum thanks for the clarification :) 
Please proceed with this PR, and we'll check internally if we keep the models in IDEAS, move them to an Obsolete package, or remove them.

**Comment by dhblum** [comment]
Thank you both @jelgerjansen and @Mathadon.  We will proceed with this PR.

**Review by dhblum — CHANGES_REQUESTED** [review]
@EttoreZ Please address my two comments.

**Comment by dhblum** [comment]
@EttoreZ Can you also address https://github.com/ibpsa/project1-boptest/issues/625 in this PR?

**Comment by EttoreZ** [comment]
> @EttoreZ Can you also address #625 in this PR?

Hey @dhblum I gave it a bit more thought. I think it makes more sense to integrate #625 with the min max PR for #658, since I am already updating the inputs for `BETEST_hydronic` and `BESTEST_hydronic_heat_pump` there, and it would be easier to merge them without conflicts. Is that ok?

**Comment by dhblum** [comment]
@EttoreZ Yes, that sounds fine.  Thanks for thinking strategically about it.  Then just let me know if/when this PR is ready.

**Comment by EttoreZ** [comment]
Ok perfect, @dhblum PR should be ready now and I will work on the min max PR.

File: testcases/bestest_hydronic_heat_pump/models/BESTESTHydronicHeatPump/TestCase.mo [hunk 1]
```diff
@@ -1,5 +1,938 @@
...
+The <b>Electricity Emissions Factor</b> profile is: 
+<ul>
+It is used a constant emission factor for electricity of 0.167 kgCO2/kWh, 
+which is the grid electricity emission factor reported by the Association of Issuing Bodies 
+(AIB) for year 2018. For reference, see: 
+<a href=\"https://www.carbonfootprint.com/docs/2019_06_emissions_factors_sources_for_2019_electricity.pdf\">
+https://www.carbonfootprint.com/docs/2019_06_emissions_factors_sources_for_2019_electricity.pdf</a> 
+</ul>
+</p>
+</html>", revisions="<html>
```
[the remark this answers is not in the document]
Comment (dhblum) [reply testcases/bestest_hydronic_heat_pump/models/BESTESTHydronicHeatPump/TestCase.mo:878]: Resolved in https://github.com/ibpsa/project1-boptest/pull/681/commits/a7cea379e586967d6f513ab29e2c6abd1b99f1d4.

File: testcases/bestest_hydronic/models/BESTESTHydronic/TestCase.mo [hunk 2]
```diff
@@ -1,5 +1,633 @@
...
+<ul>
+Based on the kgCO2 emitted per amount of natural gas burned in terms of 
+energy content.  It is 0.18108 kgCO2/kWh (53.07 kgCO2/milBTU).
+For reference,
+see:
+<a href=\"https://www.eia.gov/environment/emissions/co2_vol_mass.php\">
+https://www.eia.gov/environment/emissions/co2_vol_mass.php</a> 
+</ul>
+</p>
+</html>", revisions="<html>
```
[the remark this answers is not in the document]
Comment (dhblum) [reply testcases/bestest_hydronic/models/BESTESTHydronic/TestCase.mo:576]: Resolved in https://github.com/ibpsa/project1-boptest/pull/681/commits/a7cea379e586967d6f513ab29e2c6abd1b99f1d4.

**Pull request merged**